crates/metal-codegen-llvm/src/lib.rs

```diff
@@ -26,6 +32,31 @@ pub struct LLVMRefs {
     locals: HashMap<&'static str, LLVMValueRef>,
 }
 
+impl LLVMRefs {
+    pub fn new(module: &Module) -> Self {
+        let ctx = unsafe { LLVMContextCreate() };
+        let module_name = CString::new(module.name.as_str()).unwrap();
+        unsafe {
+            Self {
+                ctx,
+                builder: LLVMCreateBuilder(),
+                module: LLVMModuleCreateWithNameInContext(module_name.as_ptr(), ctx),
+                locals: HashMap::new(),
+            }
+        }
+    }
+}
+
+impl Drop for LLVMRefs {
```

Comment:
Drop isn't guaranteed to run, is this critical for LLVM to work or will it just result in e.g. excess memory usage?

Reply:
> Drop isn't guaranteed to run, is this critical for LLVM to work or will it just result in e.g. excess memory usage?

it prevents us from not disposing of any LLVM references/types.

Reply:
What "not disposing of any LLVM references/types" results in?

Reply:
> What "not disposing of any LLVM references/types" results in?

A memory leak since the memory for the LLVM context isn't disposed of.
And this would also scale with program size, which wouldn't be a good leak.

Reply:
Yeah just clarifying, there's not much we can do